Jenkinsfile-auto-deploy

```diff
@@ -0,0 +1,85 @@
+import groovy.json.JsonSlurperClassic
+
+def authHeader = "-H 'Authorization: token ${GITHUB_TOKEN}'"
+def contentTypeHeader = "-H 'Content-type: application/json'"
+def issueUrl = "https://api.github.com/repos/Wikia/design-system/issues/${params.pull_num}"
+def labels = ""
+def githubPagesBranch = "gh-pages"
+def publishPatchLabel = "publish-patch-on-merge"
+def publishMinorLabel = "publish-minor-on-merge"
+def publishMajorLabel = "publish-major-on-merge"
+def deployDocsLabel = "deploy-docs-on-merge"
+
+node('qa-executors') {
+	stage('Clone repo and validate') {
+		nodejs('v10 LTS') {
+			git credentialsId: 'bd3cf712-39a7-4b16-979e-ff86208ab2ea', url: 'git@github.com:Wikia/design-system.git', branch: "master"
+			currentBuild.description = "commit: ${params.commit}, PR: ${params.pull_num}"
+
+			stage('Check Pull Request labels') {
+				def json = sh(returnStdout: true, script: "curl -X GET ${authHeader} ${contentTypeHeader} ${issueUrl}/labels")
+				def parsed = new JsonSlurperClassic().parseText(json)
+				labels = parsed.name
+				println "Labels found: " + labels
+			}
+
+			withEnv(["ARTIFACTORY_NPM_TOKEN=${env.JENKINS_ARTIFACTORY_PASSWORD}"]) {
+				stage('Install dependencies') {
+					try {
+						sh 'yarn install'
+					} catch (e) {
+						error 'FAIL'
+					}
+				}
+
+				stage('Run tests') {
+					try {
+						sh 'yarn test:jenkins'
+					} catch (e) {
+						error 'FAIL'
+					}
+				}
+
+				if (currentBuild.currentResult.equals("SUCCESS")) {
+					if (labels.contains(publishMajorLabel)) {
```

Comment:
In general this is good, although I'm thinking what should happend when someone adds two or three labels. It's kind of an edge case, but might be worth considering what should happen.

Reply:
It's an edge-case but it's handled but this set of if/elseif blocks. The priorities are:
1. major
2. minor
3. patch
So if we have all 3 labels then it'll release major version.